manager/integration/tests/test_infra.py

```diff
@@ -0,0 +1,77 @@
+import os
+import pytest
+import time
+
+from digitalocean import digitalocean
+from common import get_core_api_client, get_longhorn_api_client
+from common import RETRY_COUNTS, RETRY_INTERVAL
+
+
+def detect_cloudprovider():
+    cloudprovider_name = os.getenv("CLOUDPROVIDER")
+
+    if cloudprovider_name == "digitalocean":
+        cloudprovider = digitalocean()
+        return cloudprovider
+
+
+@pytest.mark.infra
+def test_offline_node():
+    node_worker_label = 'node-role.kubernetes.io/worker'
+    pod_lable_selector = "longhorn-test=test-job"
+
+    k8s_api_client = get_core_api_client()
+    longhorn_api_client = get_longhorn_api_client()
+    cloudprovider = detect_cloudprovider()
+
+    for pod in k8s_api_client.list_namespaced_pod(
+                    'default',
+                    label_selector=pod_lable_selector).items:
+        if pod.metadata.name == "longhorn-test":
+            longhorn_test_node_name = pod.spec.node_name
+
+    for node_item in k8s_api_client.list_node().items:
+        if node_worker_label in node_item.metadata.labels and \
+                node_item.metadata.labels[node_worker_label] == 'true':
+            node_name = node_item.metadata.name
+            if node_name == longhorn_test_node_name:
+                continue
+            else:
+                break
+    node = cloudprovider.node_id(node_name)
+
+    cloudprovider.node_shutdown(node)
+
+    for i in range(RETRY_COUNTS):
+        node_status_k8s = k8s_api_client.read_node_status(node_name)
+
+        node_down = False
+        for node_condition in node_status_k8s.status.conditions:
+            if node_condition.type == "Ready" and \
+               node_condition.status != "True":
+                node_down = True
+                break
+
+        if node_down is True:
+            break
+        else:
+            time.sleep(RETRY_INTERVAL)
+            continue
+
+    for longhorn_node in longhorn_api_client.list_node():
+        if longhorn_node.name == node_name:
+            offline_node = longhorn_node
+            break
+
+    for i in range(RETRY_COUNTS):
+        longhorn_node_down = False
+        if offline_node.conditions.Ready.status == 'False':
+            longhorn_node_down = True
+            break
+        else:
+            time.sleep(RETRY_INTERVAL)
+            continue
+
+    assert longhorn_node_down
+
+    cloudprovider.node_start(node)
```

Comment:
Is it OK to run other test cases immediately after this? Should we wait for the node to be back before continuing?

It can also be problematic if something failed in this test which may result in other test failures. We can use `cloud_provider` as a fixture and add cleanup method in it to make sure everything is up even test failure happened.